tests/src/Interop/PInvoke/Generics/GenericsTest.cs

```diff
@@ -0,0 +1,206 @@
+using System.Reflection;
+using System.Text;
+using TestLibrary;
+
+unsafe class GenericsNative
+{
+    public struct Point1<T> where T : struct
+    {
+        public T x;
+    };
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point1<float> GetPoint1F();
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point1<float> AddPoint1F(Point1<float> lhs, Point1<float> rhs);
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point1<uint> GetPoint1U();
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point1<uint> AddPoint1U(Point1<uint> lhs, Point1<uint> rhs);
+
+    public struct Point2<T> where T : struct
+    {
+        public T x;
+        public T y;
+    };
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point2<float> GetPoint2F();
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point2<float> AddPoint2F(Point2<float> lhs, Point2<float> rhs);
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point2<uint> GetPoint2U();
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point2<uint> AddPoint2U(Point2<uint> lhs, Point2<uint> rhs);
+
+    public struct Point3<T> where T : struct
+    {
+        public T x;
+        public T y;
+        public T z;
+    };
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point3<float> GetPoint3F();
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point3<float> AddPoint3F(Point3<float> lhs, Point3<float> rhs);
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point3<uint> GetPoint3U();
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point3<uint> AddPoint3U(Point3<uint> lhs, Point3<uint> rhs);
+
+    public struct Point4<T> where T : struct
+    {
+        public T x;
+        public T y;
+        public T z;
+        public T w;
+    };
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point4<float> GetPoint4F();
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point4<float> AddPoint4F(Point4<float> lhs, Point4<float> rhs);
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point4<uint> GetPoint4U();
+
+    [DllImport(nameof(GenericsNative))]
+    public static extern Point4<uint> AddPoint4U(Point4<uint> lhs, Point4<uint> rhs);
+}
+
+class GenericsTest
+{
+    public unsafe static int Main(string[] args)
+    {
+        try
+        {
+            TestPoint1F();
+            TestPoint1U();
+            TestPoint2F();
+            TestPoint2U();
+            TestPoint3F();
+            TestPoint3U();
+            TestPoint4F();
+            TestPoint4U();
+        }
+        catch (System.Exception ex)
+        {
+            Console.WriteLine(ex);
+            return 0;
+        }
+        return 100;
+    }
+
+    private static void TestPoint1F()
+    {
+        GenericsNative.Point1<float> value = GenericsNative.GetPoint1F();
```

Comment:
I really appreciate the value being in the name of the function, but would prefer functions that accept a passed in value instead.  The native side should contain nothing that is 'expected' by the test.

Ideally tests should be written in a way where all data for validation is in the test itself rather than returned from external assemblies or components.

```
float v = 1.0f;
GenericsNative.Point1<float> value = GenericsNative.GetPoint1F(v);
Assert.AreEqual(value.x, v);
```

Reply:
I can certainly add some tests that do that; but I think it is just as important to test the real world scenario of: The result is created and returned by native code.

There is a lot of native code out there that is `Foo GetX` and `void GetX(Foo* pResult)`

Reply:
Agreed, but that scenario is covered by passing in a value as well. There is no need for test function that takes no arguments and one that does. At least not in this scenario.

Reply:
Fixed.